--- aegisub-3.2.2/src/colour_button.cpp.orig	2019-04-15 08:43:16.415251576 +0200
+++ aegisub-3.2.2/src/colour_button.cpp	2019-04-15 08:44:56.040476824 +0200
@@ -18,7 +18,7 @@
 
 #include "dialogs.h"
 
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil.hpp>
 
 wxDEFINE_EVENT(EVT_COLOR, wxThreadEvent);
 
--- aegisub-3.2.2/src/subtitles_provider_libass.cpp~	2014-12-08 01:07:09.000000000 +0100
+++ aegisub-3.2.2/src/subtitles_provider_libass.cpp	2019-04-15 08:49:46.265809744 +0200
@@ -50,7 +50,7 @@
 #include <libaegisub/util.h>
 
 #include <atomic>
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil.hpp>
 #include <memory>
 #include <mutex>
 
--- aegisub-3.2.2/src/video_frame.cpp~	2014-12-08 01:07:09.000000000 +0100
+++ aegisub-3.2.2/src/video_frame.cpp	2019-04-15 09:02:28.425810852 +0200
@@ -16,7 +16,7 @@
 
 #include "video_frame.h"
 
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil.hpp>
 #include <wx/image.h>
 
 namespace {
--- aegisub-3.2.2/src/video_provider_dummy.cpp~	2014-12-08 01:07:09.000000000 +0100
+++ aegisub-3.2.2/src/video_provider_dummy.cpp	2019-04-15 09:03:44.153106555 +0200
@@ -46,7 +46,7 @@
 #include <boost/algorithm/string/split.hpp>
 #include <boost/filesystem/path.hpp>
 #include <libaegisub/format.h>
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil.hpp>
 
 DummyVideoProvider::DummyVideoProvider(double fps, int frames, int width, int height, agi::Color colour, bool pattern)
 : framecount(frames)
