--- aegisub-3.2.2/tools/Makefile~	2014-12-08 01:07:09.000000000 +0100
+++ aegisub-3.2.2/tools/Makefile	2015-11-21 15:55:34.835987973 +0100
@@ -6,7 +6,7 @@
 PROGRAM += $(d)osx-bundle-restart-helper
 endif
 
-repack-thes-dict_OBJ  := $(d)repack-thes-dict.o $(TOP)lib/libaegisub.a
+repack-thes-dict_OBJ  := $(d)repack-thes-dict.o $(TOP)lib/libaegisub.a -lpthread
 repack-thes-dict_LIBS := $(LIBS_BOOST) $(LIBS_ICU)
 repack-thes-dict_CPPFLAGS := -I$(TOP) -I$(TOP)libaegisub/include $(CFLAGS_ICU)
 
